extensions/wikia/UserRenameTool/process/ProcessGlobal.class.php

```diff
@@ -0,0 +1,597 @@
+		if ( empty( $oldUserName ) ) {
+			$this->addError(
+				wfMessage( 'userrenametool-errorinvalid', $this->requestData->oldUsername )
+					->inContentLanguage()
+					->text()
+			);
+			return false;
+		}
+
+		if ( empty( $newUserName ) ) {
+			$this->addError(
+				wfMessage( 'userrenametool-errorinvalidnew', $this->requestData->newUsername )
+					->inContentLanguage()
+					->text()
+			);
+			return false;
+		}
+
+		if ( $this->areUserNamesIdentical( $oldUserName, $newUserName ) ) {
+			return false;
+		}
+
+		return true;
+	}
+
+	private function setNamesAndIds( $uid, \User $newUser, \User $oldUser ) {
+		$fakeUid = 0;
+
+		// If new user name does exist (we have a special case - repeating rename process)
+		if ( $this->newUserExists( $newUser ) ) {
+			$oldTitle = \Title::makeTitleSafe( NS_USER, $oldUser->getName() );
+			$newUserNameAlreadyExistsOut = $this->handleIfNewUserNameAlreadyExists(
+				$uid,
+				$newUser,
+				$oldUser,
+				$oldTitle
+			);
+
+			if ( !empty( $newUserNameAlreadyExistsOut ) ) {
+				list( $oldUser, $uid, $fakeUid ) = $newUserNameAlreadyExistsOut;
+			}
+		}
+
+		$this->oldUsername = $oldUser->getName();
+		$this->newUsername = $newUser->getName();
+		$this->userId = (int) $uid;
+		$this->fakeUserId = $fakeUid;
+	}
+
+	protected function newUserExists( \User $newUser ) {
+		$dbr = wfGetDB( DB_SLAVE, [], \F::app()->wg->ExternalSharedDB );
+
+		$table = \UserRenameToolHelper::getCentralUserTable();
+		$id = $dbr->selectField(
+			$table,
+			'user_id',
+			[ 'user_name' => $newUser->getName() ],
+			__METHOD__
+		);
+
+		return !empty( $id );
+	}
+
+	/**
+	 * Do the whole dirty job of renaming user
+	 *
+	 * @return bool True if the process succeeded
+	 */
+	private function renameUser() {
+		$this->logRenameStart();
+
+		// Delete the record from all the secondary clusters
+		if ( class_exists( 'ExternalUser_Wikia' ) ) {
+			\ExternalUser_Wikia::removeFromSecondaryClusters( $this->userId );
+		}
+
+		// rename the user on the shared cluster
+		if ( !$this->renameAccount() ) {
+			$this->logError( "Failed to rename the user on the primary cluster" );
+			$this->addError( wfMessage( 'userrenametool-error-cannot-rename-account' )->inContentLanguage()->text() );
+
+			return false;
+		}
+
+		$this->invalidateUserCache( $this->newUsername );
+		$this->invalidateUserCache( $this->oldUsername );
+
+		// Create a dummy account under the old username
+		$this->initializeFakeUser();
+
+		$this->updateGlobalTables();
+
+		$this->queueMultiWikiRenameTask();
+
+		return true;
+	}
+
+	private function logRenameStart() {
+		$this->logInfo( "User rename global task start." );
+		if ( !empty( $this->fakeUserId ) ) {
+			$this->logDebug( 'Process is being repeated.' );
+		};
+	}
+
+	private function renameAccount() {
+		global $wgExternalSharedDB;
+
+		$dbw = wfGetDB( DB_MASTER, [ ], $wgExternalSharedDB );
+
+		$this->logInfo( "Renaming old username to new username", [ 'cluster' => $wgExternalSharedDB ] );
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "cluster" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)